client/a8c-for-agencies/sections/marketplace/billing-dragon-checkout/index.tsx

```diff
@@ -26,24 +29,97 @@ const debug = debugFactory( 'a4a:bd-checkout' );
 function BillingDragonCheckoutContent( {
 	cartItems,
 	withA8cLogo = true,
+	siteSlug,
+	planSlug,
 }: {
 	cartItems: ShoppingCartItem[];
 	withA8cLogo?: boolean;
+	siteSlug?: string;
+	planSlug?: string;
 } ) {
 	const translate = useTranslate();
 	const [ isReady, setIsReady ] = useState( false );
 	const [ error, setError ] = useState< string | null >( null );
 
-	// Access the shopping cart API
-	const { replaceProductsInCart, responseCart } = useShoppingCart( 'no-site' );
-
+	const dispatch = useDispatch();
 	const agency = useSelector( getActiveAgency );
+	const site = useSelector( ( state ) => ( siteSlug ? getSite( state, siteSlug ) : undefined ) );
+	const siteId = site?.ID;
+	const isPlanCheckout = !! planSlug;
+
+	useEffect( () => {
+		if ( siteId ) {
+			dispatch( setSelectedSiteId( siteId ) );
+		}
+	}, [ dispatch, siteId ] );
+
+	// Use site's cart key when site exists, otherwise use 'no-site' for siteless checkout
+	const cartKey = siteId || 'no-site';
+	const { replaceProductsInCart, responseCart } = useShoppingCart( cartKey );
+
+	const {
+		productsForCart,
+		isLoading: areProductsPreparing,
+		error: productsError,
+	} = usePrepareProductsForCart( {
+		productAliasFromUrl: planSlug,
+		purchaseId: null,
+		usesJetpackProducts: false,
+		isPrivate: false,
+		siteSlug,
+		sitelessCheckoutType: 'a4a',
+	} );
 
 	debug( '[A4A Checkout] Cart items: ', cartItems );
 
+	useEffect( () => {
```

Comment:
It be worth adding a descriptive comment here to explain when this is used, then maybe also  for the existing flow to make it easier to know when the too flows are used.